### PR TITLE
Migration guide for package:flutter_lints

### PR DESCRIPTION
**Pull request by goderbauer**
Pre-work for this PR is done:
- [x] [package:flutter_lint](https://pub.dev/packages/flutter_lints) has published a stable version 1.0.0
- [x] https://github.com/flutter/flutter/pull/81417 is merged
- [x] This PR is updated with the Flutter version number which contains https://github.com/flutter/flutter/pull/81417

### CODE REVIEW

**Comment by goderbauer** [comment]
FYI @pq @mit-mit @devoncarew @csells

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+---
+title: Introducing package:flutter_lints
+description: Migrate to package:flutter_lints to get the latest set of recommended lints, which encourage good coding practices.
+---
+
+## Summary
+
+The [`package:flutter_lints`][] defines the latest set of recommended lints that
+encourage good coding practises for Flutter apps, packages, and plugins. Projects
```
Comment (pq) [inline src/docs/release/breaking-changes/flutter-lints-package.md:9]: I *think* this should be "practices".

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+created with `flutter create` after Flutter version 2.xx are already enabled to
+use the latest set of recommended lints. Projects created prior to that version
+can upgrade to it with the instructions in this guide.
+
+## Context
+
+Prior to the introduction of `package:flutter_lints` the Flutter framework
```
Comment (pq) [inline src/docs/release/breaking-changes/flutter-lints-package.md:16]: =>

Prior to the introduction of `package:flutter_lints`,

?

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+shipped with a set of lints defined in [`analysis_options_user.yaml`][] that was
+used by the [dart analyzer][] to identify code issues if a Flutter project
+didn't define a custom `analysis_options.yaml` file. Since `analysis_options_user.yaml`
+was tied to a particular framework version, it was difficult to evolve without
+breaking existing apps, packages, and plugins. As a result of that the lints
```
Comment (pq) [inline src/docs/release/breaking-changes/flutter-lints-package.md:21]: => 

As a result of that, 

?

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+defined in `analysis_options_user.yaml` are heavily outdated. To fix this,
+`package:flutter_lints` was created. The package versions the lint set to enable
+evolving it without breaking existing projects. Since the package builds on
+Dart's [`package:lints`][] it also aligns the lints recommended for Flutter
+projects with the rest of the dart universe.
```
Comment (pq) [inline src/docs/release/breaking-changes/flutter-lints-package.md:26]: => 

Dart universe.

?

**Review by pq — COMMENTED** [review]
In general, looks great.  Nice docs!

A few super minor nits...

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+shipped with a set of lints defined in [`analysis_options_user.yaml`][] that was
+used by the [dart analyzer][] to identify code issues if a Flutter project
+didn't define a custom `analysis_options.yaml` file. Since `analysis_options_user.yaml`
+was tied to a particular framework version, it was difficult to evolve without
+breaking existing apps, packages, and plugins. As a result of that, the lints
+defined in `analysis_options_user.yaml` are heavily outdated. To fix this,
+`package:flutter_lints` was created. The package versions the lint set to enable
+evolving it without breaking existing projects. Since the package builds on
+Dart's [`package:lints`][] it also aligns the lints recommended for Flutter
+projects with the rest of the Dart universe.
```
Comment (mit-mit) [inline src/docs/release/breaking-changes/flutter-lints-package.md:26]: Suggest s/universe/ecosystem/ which sounds a bit friendlier?

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+by running `flutter pub add --dev flutter_lints` in the root directory of the
+project.
+
+Create an `analysis_options.yaml` file in the root directory of your project
+(next to the `pubspec.yaml` file) with the following content:
+```yaml
+include: package:flutter_lints/flutter.yaml
+```
+
+The newly activated lint set may identify some previously undiscovered issues in
```
Comment (mit-mit) [inline src/docs/release/breaking-changes/flutter-lints-package.md:43]: s/some previously undiscovered/new/ ?

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+your code. To find them, run `flutter analyze` in the root directory of your
```
Comment (mit-mit) [inline src/docs/release/breaking-changes/flutter-lints-package.md:44]: Perhaps also mention that you can simply open the project in an IDE with Dart support?

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+project.
+
+### Existing custom analyzer_options.yaml file
+
+If your project already has a custom `analysis_options.yaml` file at its root,
+add `include: package:flutter_lints/flutter.yaml` to it at the top to activate
+the lints from `package:flutter_lints`. If your `analysis_options.yaml` already
+contains an `include:` directive you have to decide whether you want to keep
+those lints or whether you want to replace it with the lints from
+`package:flutter_lints` because the dart analyzer only supports one `include:`
```
Comment (mit-mit) [inline src/docs/release/breaking-changes/flutter-lints-package.md:56]: s/dart/Dart/

File: src/docs/release/breaking-changes/flutter-lints-package.md (hunk 1 above)
Comment (sfshaza2) [reply src/docs/release/breaking-changes/flutter-lints-package.md]: Correct, we use American English, not British English on our developer sites:  https://developers.google.com/style/spelling?hl=en. :D

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+* [Add flutter_lints package][]
+* [Integrate package:flutter_lints into templates][]
+
+[Add flutter_lints package]: {{site.github}}/flutter/packages/pull/343
+[`analysis_options_user.yaml`]: {{site.github}}/flutter/flutter/blob/master/packages/flutter/lib/analysis_options_user.yaml
+[Customizing static analysis]: https://dart.dev/guides/language/analysis-options
+[dart analyzer]: https://dart.dev/guides/language/analysis-options
+[Integrate package:flutter_lints into template]: {{site.github}}/flutter/flutter/pull/81417
+[Issue 78432 - Update lint set for Flutter applications]: {{site.github}}/flutter/flutter/issues/78432
+[`package:flutter_lints`]: https://pub.dev/packages/flutter_lints
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/flutter-lints-package.md:126]:
```suggestion
[`package:flutter_lints`]: {{site.pub}}/packages/flutter_lints
```

File: src/docs/release/breaking-changes/flutter-lints-package.md [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+    # avoid_print: false  # Uncomment to disable the `avoid_print` rule
+    # prefer_single_quotes: true  # Uncomment to enable the `prefer_single_quotes` rule
+
+# Additional information about this file can be found at
+# https://dart.dev/guides/language/analysis-options
+```
+
+## Timeline
+
+Landed in version: 2.xx<br>
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply src/docs/release/breaking-changes/flutter-lints-package.md:100]: Not yet. I'll update this PR as soon as I know.

File: src/docs/release/breaking-changes/index.md [hunk 11]
```diff
@@ -13,9 +13,11 @@ release, and listed in alphabetical order:
 ### Not yet released to stable
 
 * [Default Scrollbars on Desktop][]
+* [Introducing package:flutter_lints][]
 * [ThemeData's accent properties have been deprecated][]
 
 [Default Scrollbars on Desktop]: /docs/release/breaking-changes/default-desktop-scrollbars
+[Introducing package:flutter_lints]: /docs/release/breaking-changes/flutter-lints-package
```
[the remark this answers is not in the document]
Comment (goderbauer) [reply src/docs/release/breaking-changes/index.md:20]: I am open to that. :) Do you have a place in mind?

Also, we will link to this page from tooling message when it is discovered that your setup is out of date.

**Review by sfshaza2 — COMMENTED** [review]
On second thought, I'm ok with this living in the breaking changes docs...

**Comment by goderbauer** [comment]
I am now just waiting on a Flutter dev release that contains https://github.com/flutter/flutter/pull/81417 to obtain the version number that needs to be included in this guide.

**Comment by sfshaza2** [comment]
SG

**Comment by goderbauer** [comment]
Added the missing version info. This PR is ready now!

**Review by pq — APPROVED** [review]
Nice doc!

**Review by sfshaza2 — APPROVED** [review]
LGTM

**Pull request merged**